Server/mods/deathmatch/logic/CMainConfig.cpp

```diff
@@ -847,44 +856,87 @@ bool CMainConfig::AddMissingSettings()
-    if (!pRootNodeTemplate)
+    CXMLFile* templateFile = g_pServerInterface->GetXML()->CreateXML(templateFileName);
+    CXMLNode* templateRootNode = templateFile && templateFile->Parse() ? templateFile->GetRootNode() : nullptr;
+    if (!templateRootNode)
     {
-        CLogger::ErrorPrintf("Can't parse '%s'\n", *strTemplateFilename);
+        CLogger::ErrorPrintf("Can't parse '%s'\n", *templateFileName);
         return false;
     }
 
     // Check that each item in the template also exists in the server config
-    bool      bChanged = false;
-    CXMLNode* pPrevNode = nullptr;
-    for (auto it = pRootNodeTemplate->ChildrenBegin(); it != pRootNodeTemplate->ChildrenEnd(); ++it)
-    {
-        CXMLNode* pNodeTemplate = *it;
-        SString   strNodeName = pNodeTemplate->GetTagName();
-        CXMLNode* pNode = m_pRootNode->FindSubNode(strNodeName);
-        if (!pNode)
+    bool      hasConfigChanged = false;
+    CXMLNode* previousNode = nullptr;
+    for (auto it = templateRootNode->ChildrenBegin(); it != templateRootNode->ChildrenEnd(); ++it)
+    {
+        CXMLNode* templateNode = *it;
+        SString   templateNodeTagName = templateNode->GetTagName();
+
+        // Find node with exact same attributes
+        CXMLAttributes& templateAttributes = templateNode->GetAttributes();
+        CXMLNode*       foundNode = nullptr;
+        for (auto it2 = m_pRootNode->ChildrenBegin(); it2 != m_pRootNode->ChildrenEnd(); ++it2)
         {
-            CLogger::LogPrintf("Adding missing '%s' to mtaserver.conf\n", *strNodeName);
-            SString strNodeValue = pNodeTemplate->GetTagContent();
-            SString strNodeComment = pNodeTemplate->GetCommentText();
-            pNode = m_pRootNode->CreateSubNode(strNodeName, pPrevNode);
-            pNode->SetTagContent(strNodeValue);
-            pNode->SetCommentText(strNodeComment, true);
-            bChanged = true;
+            CXMLNode* tempNode = *it2;
+            if (tempNode->GetTagName() == templateNodeTagName)
+            {
+                bool bAttributesMatch = true;
+                CXMLAttributes& attributes = tempNode->GetAttributes();
+                for (auto it3 = templateAttributes.ListBegin(); it3 != templateAttributes.ListEnd(); ++it3)
+                {
+                    CXMLAttribute* templateAttribute = *it3;
+                    const SString& strKey = templateAttribute->GetName();
+                    const SString& strValue = templateAttribute->GetValue();
+                    CXMLAttribute* foundAttribute = attributes.Find(strKey);
+                    if (!foundAttribute || foundAttribute->GetValue() != strValue)
+                    {
+                        bAttributesMatch = false;
+                        break;
+                    }
+                }
+                if (bAttributesMatch)
+                {
+                    foundNode = tempNode;
+                    break;
+                }
```

Comment:
use early continue

Reply:
There is no early continue practice. Leave previous variant.

Reply:
> There is no early continue practice. Leave previous variant.

Yes there is.

Reply:
> Yes there is.

Prove it

Reply:
> > Yes there is.
> 
> Prove it

https://github.com/multitheftauto/mtasa-docs/blob/main/mtasa-blue/CODING_GUIDELINES.md#code-readability

Reply:
It's  `early-returns` and not `early continue`

Reply:
> It's `early-returns` and not `early continue`

My friend, look below that

Reply:
It's your own rule, lol

Reply:
> It's your own rule, lol

Its been merged nonetheless.

Reply:
It changes almost nothing in terms of readability tbh. But ye it was approved